Tasks/CordovaBuild/lib/cordova-task.js

```diff
@@ -35,42 +35,49 @@ processInputs()															// Process inputs to task
     return targetEmulator ? 0 : teambuild.packageProject(platform);		// Package apps if configured
 })
     .then(copyToOutputFolder)
-    .then(function (code) {												// On success, exit
-    taskLibrary.exit(code);
+    .fail(function (err) {
+    var promise = deleteKeychain ? deleteKeychain() : Q(0);	        // Delete temp keychain if created
+    if (deleteProfile) {											// Delete installed profile only if flag is set
+        promise = promise.then(function (code) {
+            return deleteProfile();
+        });
+    }
+
+    console.error(err);
+    return promise.then(function () {
+        process.exit(1);
+    });
 })
-    .fin(function (code) {
+    .done(function (code) {
```

Comment:
The relevant code was put in both this done and the error handler. This was done in part to fix the temp keychains/profiles not being cleaned up. Also, since for error, we need to call process.exit with a non zero status code to propagate the fact that the task failed.